inginious/frontend/user_manager.py

```diff
@@ -510,6 +510,10 @@ def update_user_stats(self, username, task, submission, result_str, grade, newsu
             old_submission = self._database.user_tasks.find_one_and_update(
                 {"username": username, "courseid": submission["courseid"], "taskid": submission["taskid"]}, {"$inc": {"tried": 1, "tokens.amount": 1}})
 
+            # If this is a submission after deadline, do not affect the final grade.
+            if submission.get("is_later_submission", False):
+                return
+
```

Comment:
As this is checked in both cases (if and else), can it be out of the if statement?

Reply:
No, the thing is that the logic is different in the `if` and the `else`, the position of this check affects in a different way the logic.